src/reducers/Material.js

```diff
@@ -19,6 +19,7 @@ function materialsUpdateOne(state, action) {
     const index = action.index || state.index;  // not passing index when modifying currently displayed material
     const material = action.material.clone();   // clone material to assert props re-render
     material.isUpdated = true;                  // to be used inside components
+    material.isNonPeriodic = action.material.isNonPeriodic;
```

Comment:
Without this line the state does not retain the `isNonPeriodic` value

![Screen Shot 2021-11-29 at 7 50 18 PM](https://user-images.githubusercontent.com/23506416/143983081-db199676-80cb-4739-921f-8552b515b0c7.png)


With the line the value is retained

![Screen Shot 2021-11-29 at 7 56 01 PM](https://user-images.githubusercontent.com/23506416/143983140-4cbba634-627f-4c2c-9069-b6dc13632bbe.png)

Reply:
Then we should look into the `clone()` function (or Material class itself and its `toJSON`) and figure out why it doesn't retain the value

Reply:
I will do that.